### PR TITLE
doc: Document release process

### PR DESCRIPTION
**Pull request by DevinCodes**
| Q                 | A
| ----------------- | ----------
| Bug fix?          | no
| New feature?      | no

## Description
This PR adds documentation on how to do a release for Frontman. Only administrators can do a release, but documenting the process publicly allows people to discuss, improve on it, or even automate most of the steps.

### CODE REVIEW

File: .github/_release.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+ # Frontman release process
+ 
+ ## Setup
+ 
+ > You only have to do this setup once.
```
Comment (sarahdayan) [inline .github/_release.md:5]:
```suggestion
 > You only have to do this once.
```

File: .github/_release.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ 
+ * Make sure you have an account on [RubyGems][rubygems].
+ * Run the `gem signin` command from your CLI, and log in with your RubyGems credentials.
```
Comment (sarahdayan) [inline .github/_release.md:8]:
```suggestion
 * Run the `gem signin` command from your CLI and log in with your RubyGems credentials.
```

File: .github/_release.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+ * Make sure that you're an administrator on the Frontman gem. You can ask an existing administrator for access if you're in a position to do releases for Frontman.
```
Comment (sarahdayan) [inline .github/_release.md:9]:
```suggestion
 * Make sure you're an administrator on the [Frontman][frontman] gem. You can ask an existing administrator for access if you're in a position to release Frontman.
```

File: .github/_release.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+ * Commit your changes: `git commit -a -m "chore: bumps version to vX.X.X"`.
+ * Create a tag for the new version: `git tag X.X.X`.
+ * Push your changes and the new tag: `git push && git push --tags`.
+ * Build the Frontman gem: `gem build frontman-ssg.gemspec`.
+ * Push the newly built gem to [RubyGems][rubygems]: `gem push frontman-ssg-X.X.X.gem`.
+ * Delete the locally built gem: `rm frontman-ssg-X.X.X.gem`.
+ 
+ 
+ [changelog]: ../CHANGELOG.md
+ [rubygems]: https://www.rubygems.org/
```
Comment (sarahdayan) [inline .github/_release.md:29]:
```suggestion
[frontman]: https://rubygems.org/gems/frontman-ssg
[rubygems]: https://www.rubygems.org/
[semver]: https://semver.org/
```

File: .github/_release.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+ 
+ * Make sure you have an account on [RubyGems][rubygems].
+ * Run the `gem signin` command from your CLI, and log in with your RubyGems credentials.
+ * Make sure that you're an administrator on the Frontman gem. You can ask an existing administrator for access if you're in a position to do releases for Frontman.
+ 
+ ## Release 
+ * First, make sure all the tests on `master` are green.
+ * Switch to the `master` branch: `git checkout master`.
+ * Make sure you branch is up-to-date: `git fetch && git pull`.
+ * Make sure your repository is clean: `git reset --hard`.
```
Comment (sarahdayan) [inline .github/_release.md:15]: `git reset --hard` leaves untracked files.

```suggestion
 * Make sure your repository is clean: `git reset --hard && git clean -dfx`.
```

File: .github/_release.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+ * Check the changes the new version introduces by visiting https://github.com/algolia/frontman/compare/{latest_version}...master , where you have to replace `{latest version}` with the latest tag. You can use `git tag --list` to check for the latest release.
```
Comment (sarahdayan) [inline .github/_release.md:16]:
```suggestion
 * Check the changes the new version introduces by visiting `github.com/algolia/frontman/compare/{latest_version}...master`, replacing `{latest version}` with the latest tag. You can run `git tag --list` to check for the latest release.
```

File: .github/_release.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+ * Determine the new version number that you should release, and make sure there are no breaking changes.
```
Comment (sarahdayan) [inline .github/_release.md:17]:
```suggestion
 * Determine the [new version number][semver] that you should release, and make sure there are no breaking changes.
```

File: .github/_release.md [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+ * Bump the version in `lib/frontman/version.rb`.
+ * Update the [`CHANGELOG.md`][changelog] file with the changes that this new version introduces.
+ * Commit your changes: `git commit -a -m "chore: bumps version to vX.X.X"`.
```
Comment (sarahdayan) [inline .github/_release.md:20]:
```suggestion
 * Commit your changes: `git commit -a -m "chore: bumps version to vX.Y.Z"`.
```

File: .github/_release.md [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+ * Create a tag for the new version: `git tag X.X.X`.
```
Comment (sarahdayan) [inline .github/_release.md:21]:
```suggestion
 * Create a tag for the new version: `git tag X.Y.Z`.
```

File: .github/_release.md [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+ * Push your changes and the new tag: `git push && git push --tags`.
+ * Build the Frontman gem: `gem build frontman-ssg.gemspec`.
+ * Push the newly built gem to [RubyGems][rubygems]: `gem push frontman-ssg-X.X.X.gem`.
```
Comment (sarahdayan) [inline .github/_release.md:24]:
```suggestion
 * Push the newly built gem to [RubyGems][rubygems]: `gem push frontman-ssg-X.Y.Z.gem`.
```

File: .github/_release.md [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+ * Delete the locally built gem: `rm frontman-ssg-X.X.X.gem`.
```
Comment (sarahdayan) [inline .github/_release.md:25]:
```suggestion
 * Delete the locally built gem: `rm frontman-ssg-X.Y.Z.gem`.
```

File: .github/_release.md [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+
```
Comment (sarahdayan) [inline .github/_release.md:26]:
```suggestion
```

File: .github/_release.md [hunk 13]
```diff
@@ -0,0 +1,29 @@
+ # Frontman release process
```
Comment (sarahdayan) [inline .github/_release.md:1]: I'd rename the file `RELEASE.md` for consistency with the other files.

Comment (DevinCodes) [reply .github/_release.md]: Done ✅

File: .github/RELEASE.md [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+ * Bump the version in `lib/frontman/version.rb`.
+ * Update the [`CHANGELOG.md`][changelog] file with the changes that this new version introduces.
+ * Commit your changes: `git commit -a -m "chore: bumps version to vX.Y.Z"`.
+ * Create a tag for the new version: `git tag X.Y.Z`.
+ * Push your changes and the new tag: `git push && git push --tags`.
+ * Build the Frontman gem: `gem build frontman-ssg.gemspec`.
+ * Push the newly built gem to [RubyGems][rubygems]: `gem push frontman-ssg-X.Y.Z.gem`.
+ * Delete the locally built gem: `rm frontman-ssg-X.Y.Z.gem`.
+ 
+ [changelog]: ../CHANGELOG.md
```
Comment (DevinCodes) [inline .github/RELEASE.md:27]:
```suggestion
[changelog]: ../CHANGELOG.md
```

**Review by sarahdayan — APPROVED** [review]

**Pull request merged**